File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    function inorderTraversal($root) {
        
        $ans = [];
        $stack = [];
        $cur = $root;
        
        
        while($cur || count($stack) > 0) {
            if($cur) {
                array_push($stack, $cur);
                $cur = $cur->left;
            } else {
                $cur = array_pop($stack);
                array_push($ans, $cur->val);
                $cur = $cur->right;
            }
        }  
        
        return $ans;
    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

type Res struct {
    Res []int
}

func inorderTraversal(root *TreeNode) []int {
    ans := Res{}
    ans.dfs(root)
    return ans.Res
}

func (r *Res) dfs (node *TreeNode) {
    
    if node == nil {
        return
    }
    
    r.dfs(node.Left)
    r.Res = append(r.Res, node.Val)
    r.dfs(node.Right)
}